src/com/seafile/seadroid2/ui/WidgetUtils.java

```diff
@@ -101,37 +99,31 @@ public static void showFile(final FragmentActivity activity, File file) {
         String name = file.getName();
         String suffix = name.substring(name.lastIndexOf('.') + 1).toLowerCase();
 
-        if (suffix.length() == 0) {
-            ToastUtils.show(activity, R.string.unknown_file_type);
-            return;
-        }
-
         // Open markdown files in MarkdownActivity
         if (suffix.equals("md") || suffix.equals("markdown")) {
             startMarkdownActivity(activity, file.getPath());
-            activity.finish();
             activity.overridePendingTransition(0, 0);
             return;
         }
 
         String mime = MimeTypeMap.getSingleton().getMimeTypeFromExtension(suffix);
+        if (mime==null)
+            mime = "*/*"; // forces app chooser dialog on unknown type
         Intent open = new Intent(Intent.ACTION_VIEW);
+        open.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         open.setDataAndType((Uri.fromFile(file)), mime);
 
+        if (activity.getPackageManager().resolveActivity(open, 0) == null) {
+            ToastUtils.show(activity, "Could not find suitable app for mime type " + mime);
```

Comment:
better to extract this message into strings.xml for the sake of internationalization.